spring4/src/test/java/org/jdbi/v3/spring4/TestJdbiTemplate.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.spring4;
+
+import org.jdbi.v3.core.Jdbi;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.sql.DataSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {TestJdbiTemplate.Config.class})
+public class TestJdbiTemplate {
+
+    @Autowired
+    private JdbiOperations jdbiOps;
+
+    @Test
+    public void testJdbiOps() {
+        assertThat(jdbiOps).isNotNull().isInstanceOf(JdbiTemplate.class);
+    }
+
+    @Configuration
+    @EnableTransactionManagement
+    public static class Config {
+
+        @Bean
+        public DataSource dataSource() {
+            return new EmbeddedDatabaseBuilder().setType(EmbeddedDatabaseType.H2).build();
+        }
+
+        @Bean
+        public Jdbi jdbi(DataSource dataSource) {
+            return Jdbi.create(dataSource);
+        }
+
+        @Bean
+        public JdbiOperations jdbiOperations(Jdbi jdbi) {
+            return new JdbiTemplate(jdbi);
+        }
+
+    }
```

Comment:
It would be nice to add a test how to use Dao class with Spring transactions and `JdbiTemplate`. Something like:
```java
        @Bean
        public Dao dao(JdbiOperations jdbiOperations) {
            return new Dao(jdbiOperations);
        }

    }

    public static class Dao {

        private JdbiOperations jdbiOperations;

        public Dao(JdbiOperations jdbiOperations) {
            this.jdbiOperations = jdbiOperations;
        }

        @Transactional
        public Integer getAmount() {
            return jdbiOperations.withHandle(h -> h.createQuery("select count(*) from something")
                    .mapTo(Integer.class)
                    .findOnly());
        }

        @Transactional
        public void insert(int id, String name) {
            jdbiOperations.useHandle(h -> h.createUpdate("insert into something (id, name) values (?, ?)")
                    .bind(0, id)
                    .bind(1, name)
                    .execute());
        }

        @Transactional
        public void insertMany(List<Integer> ids, List<String> names) {
            for (int i = 0; i < ids.size(); i++) {
                insert(ids.get(i), names.get(i));
            }
        }
    }

    @Test
    public void testTransactionsWork() {
        dao.insertMany(Arrays.asList(1, 2, 3), Arrays.asList("Test1", "Test2", "Test3"));
        System.out.println(dao.getAmount());
    }
```

Reply:
I will add checks to verify the behaviour I expect from the plugin. this is WIP